common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1622 @@
+				offset = bytesWritten;
+				length = obj.bytesProcessed - bytesWritten;
+			}
+		} catch (e) {
+			return callback(e);
+		}
+	}, COPY_FILE_CHUNK_SIZE, true);
+}
+
+/**
+ * @param {string|Buffer|URL} path file path
+ * @param {Ti.Filesystem.FileStream} fileStream file stream
+ * @returns {integer} file descriptor
+ */
+function createFileDescriptor(path, fileStream) {
+	const pointer = fileDescriptorCount++; // increment global counter
+	const fd = new FileDescriptor(pointer, path, fileStream);
+	fileDescriptors.set(pointer, fd); // use it to refer to this file stream as the "descriptor"
+	return pointer;
+}
+
+/**
+ * @param {integer} fd file descriptor
+ * @returns {Ti.Filesystem.FileStream} matching stream
+ */
+function streamForDescriptor(fd) {
+	const wrapper = fileDescriptors.get(fd);
+	return wrapper.stream;
+}
+
+/**
+ * @param {integer} fd file descriptor
+ * @returns {string} matching stream
+ */
+function pathForFileDescriptor(fd) {
+	const wrapper = fileDescriptors.get(fd);
+	return wrapper.path;
+}
+
+/**
+ * Used to merge the user-supplied options with the defaults for a function. Special cases a string to be encoding.
+ * @param {*} options user-supplied options
+ * @param {object} defaults defaults to use
+ * @return {object}
+ */
+function mergeDefaultOptions(options, defaults) {
+	if (options === null) {
+		return defaults;
+	}
+
+	const optionsType = typeof options;
+	switch (optionsType) {
+		case 'undefined':
+		case 'function':
+			return defaults;
+		case 'string':
+			// Use copy of defaults but with encoding set to the 'options' value!
+			const merged = Object.assign({}, defaults);
+			merged.encoding = options;
+			return merged;
+		case 'object':
+			return options;
+		default:
+			assertArgumentType(options, 'options', 'object');
+			return null; // should never get reached
+	}
+}
+
+/**
+ * returns randomly generated characters of given length 1-16
+ * @param {integer} length 1 - 16
+ * @returns {string}
+ */
+function randomCharacters(length) {
+	return (Math.random().toString(36) + '00000000000000000').slice(2, length + 2);
+}
+
+function makeError(code, message, errno, syscall, path) {
+	const error = new Error(`${code}: ${message}, ${syscall} '${path}'`);
+	error.errno = errno;
+	error.syscall = syscall;
+	error.code = code;
+	error.path = path;
+	return error;
+}
+
+/**
+ * @param {string} encoding what we're encoding to
+ * @param {Ti.Buffer} tiBuffer Ti.Buffer instance
+ * @returns {Buffer} node-compatible Buffer instance
+ */
+function encodeBuffer(encoding, tiBuffer) {
+	const buffer = Buffer.from(tiBuffer);
+	switch (encoding) {
+		case 'buffer':
+		case null:
+		case undefined:
+			return buffer;
+		default:
+			return buffer.toString(encoding);
+	}
+}
+
+/**
+ * @param {string|Buffer|URL} path file path
+ * @return {Ti.Filesystem.File}
+ */
+function getTiFileFromPathLikeValue(path) {
+	// Android can't properly handle file: URIs if they don't have trailing '//'
```

Comment:
It looks like you have actually fixed that in the Android implementation, right?